src/script/ir_builder/relax/utils.h

```diff
@@ -63,6 +73,51 @@ inline BlockFrame CheckBlockFrameExistAndUnended() {
   return new_block_frame;
 }
 
+inline tvm::relax::SeqExpr GetSeqExprForBranch(const SeqExprFrame& frame, String* var_name) {
+  // Step 0. Check frame type
+  std::string method;
+  if (frame->IsInstance<ThenFrameNode>()) {
+    method = "R.Then";
+  } else if (frame->IsInstance<ElseFrameNode>()) {
+    method = "R.Then";
+  } else {
+    ICHECK(false) << "TypeError: Unsupported frame type: " << frame->GetTypeKey();
+  }
+
+  // Step 1. Check non-empty block and last binding is non-dataflow
```

Comment:
```suggestion
  // Step 1. Check non-empty block
```
Looks like we are not checking if the last binding in this block is dataflow or not in step 1, so let’s update the comment here.

Actually, since we don’t allow control flow in dataflow blocks, all if-frames must not under a dataflow block, and hence every binding variable should not be DataflowVar.

On one hand, it’s good to check this for every binding variable inside each branch, but on the other hand, following our parser behavior, these binding variables have no way to be a DataflowVar, so I think it’s reasonable not to check this. Besides, it seems to me that doing the check needs quite a lot efforts, since we will need to check for each single binding. From this angle, I also prefer not checking.